src/EventStore.ClientAPI/ClusterDnsSeed.cs

```diff
@@ -0,0 +1,36 @@
+using System.Net;
+using EventStore.ClientAPI.Common.Utils;
+using EventStore.ClientAPI.Transport.Http;
+
+namespace EventStore.ClientAPI {
+	public class ClusterDnsSeed: IGossipSeed {
+		private readonly string _clusterDns;
+		private readonly bool _seedOverTls;
+		private readonly bool _v20HostHeader;
+		private readonly int _port;
+
+		public ClusterDnsSeed(string clusterDns, int port, bool seedOverTls = false, bool v20HostHeader = false) {
+			_clusterDns = clusterDns;
+			_seedOverTls = seedOverTls;
+			_v20HostHeader = v20HostHeader;
+			_port = port;
+		}
+
+		public string ToHttpUrl() {
+			if (_v20HostHeader) {
```

Comment:
Does it make sense to call this `_v20HostHeader` ? It clearly does more than that (skips DNS resolution & returns host headers).  Rename or split into two to make it clearer would be my recommendation.

Reply:
After applying @shaan1337 last remark, it does more than what I intended initially, will find a better name.